### PR TITLE
Update proposed use case diagram

### PR DESCRIPTION
**Pull request by jpl-jengelke**
Please review with feedback. 

Graphic is available here: https://jpl-jengelke.github.io/slim/continuous-testing/use-cases/

### CODE REVIEW

File: continuous-testing/use-cases/markmap.md [hunk 1]
```diff
@@ -1,29 +1,96 @@
...
+        * For Code Quality (unified, both dynamic and static analyses)
+            * Unified analysis of static and dynamic code metrics
+            * Tools
+                * SonarQube
+                * Coverity
+        * For Security
+            * Verify and report vulnerabilities
+            * Check code security
+            * Tools
+                * Semmle
```
Comment (jeffreypon) [inline continuous-testing/use-cases/markmap.md:90]: CAE isn't supporting Semmle anymore. I'm not sure why. Microsoft Github did buy them out but it still appears to be offered through open source github. Regardless, JPL is migrating to the usage of scrub now (https://github.com/nasa/scrub).  Suggest adding scrub to the security tool list.

**Review by riverma — CHANGES_REQUESTED** [review]
Hi @jpl-jengelke -

Great work making such a comprehensive use case list! It's very thorough.

Some stylistic changes requested:
- Use "Using ..." prefix for all tool references at the leaf nodes of the use case lists
- Include hyperlinks to point to the tool homepages, for easy lookup when people read / view the generated diagram

Some general thoughts:
- It's better to keep the list of tools fewer than comprehensive - in SLIM, we're not trying to list every possible tool solution to a use case problem (that would never end), rather, we're hoping to create a curated list of _best practices_. A general question to consider is: would you stand by and vet a given tool referral to a colleague of yours? If not - it's probably not been sufficiently tested to recommend it as a "best practice". 
- For "non-tools" that are recommended in your leaf nodes, e.g. "Failure profiles", something you may want to consider is that the leaf nodes are meant to be "Actionable". In other words, they should either hyperlink to tool homepages, starter kits, or external reference standards. Please take a look at all your leaf nodes and think about whether the reader can actually _do_ something with the information provided.

File: continuous-testing/use-cases/markmap.md (hunk 1 above)
Comment (jpl-jengelke) [reply continuous-testing/use-cases/markmap.md:90]: Thanks. Will do.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# auto-generated files
+**/markmap.html
```
[the remark this answers is not in the document]
Comment (jpl-jengelke) [reply .gitignore:2]: Got it. I will make the change.

File: continuous-testing/use-cases/markmap.md [hunk 3]
```diff
@@ -35,22 +102,50 @@
...
                 * Cucumber* (Java + others) / Behave (Python) / Specflow (C#)
                 * Watir (Ruby)
                 * Home Grown
-    * For Backend (Load testing?)
-        * Using Chaos Monkey
-        * Using Robot Framework
-        * Using Cucumber Open
-        * For Mobile
+    * For Load testing
+        * Failure profiles
```
[the remark this answers is not in the document]
Comment (jpl-jengelke) [reply continuous-testing/use-cases/markmap.md:118]: Reworked entire diagram with these suggestions. Thanks.

[the remark this answers is not in the document]
Comment (jpl-jengelke) [reply continuous-testing/use-cases/markmap.md:118]: Reworked entire diagram with these suggestions. Thanks.

File: continuous-testing/use-cases/markmap.md [hunk 4]
```diff
@@ -1,29 +1,96 @@
...
-        * Unit Testing
-            * xUnit (e.g. Junit\[Java\], PyUnit\[Python\], NUnit\[C#\], CppUnit\[C++\]
-            * Tools
-                * Java
-                * Python
-                * C#
-                * C/C++
-        * Integration, Dependency and Stub Mocking
+        * For Unit Testing, Integration, Dependency and Stub Mocking
+            * xUnit for build-time testing
```
[the remark this answers is not in the document]
Comment (jpl-jengelke) [reply continuous-testing/use-cases/markmap.md:4]: Reworked entire diagram with these suggestions. Thanks.

File: continuous-testing/use-cases/markmap.md [hunk 5]
```diff
@@ -35,22 +102,50 @@
...
+        * Resource requirements
+        * Maximize performance
+        * Tools
+            * Apache JMeter
+            * Chaos Monkey
+            * Robot Framework
+            * Cucumber Open
+    * For Mobile
+        * UI and viewability
+        * Accessibility
```
[the remark this answers is not in the document]
Comment (jpl-jengelke) [reply continuous-testing/use-cases/markmap.md:128]: Reworked entire diagram with these suggestions. Thanks.

File: continuous-testing/use-cases/markmap.md [hunk 6]
```diff
@@ -35,22 +102,50 @@
...
                 * Watir (Ruby)
                 * Home Grown
-    * For Backend (Load testing?)
-        * Using Chaos Monkey
-        * Using Robot Framework
-        * Using Cucumber Open
-        * For Mobile
+    * For Load testing
+        * Failure profiles
+        * Resource requirements
```
[the remark this answers is not in the document]
Comment (jpl-jengelke) [reply continuous-testing/use-cases/markmap.md:119]: Reworked entire diagram with these suggestions. Thanks.

File: continuous-testing/use-cases/markmap.md [hunk 7]
```diff
@@ -35,22 +102,50 @@
...
+        * Maximize performance
+        * Tools
+            * Apache JMeter
```
[the remark this answers is not in the document]
Comment (jpl-jengelke) [reply continuous-testing/use-cases/markmap.md:122]: Reworked entire diagram with these suggestions. Thanks.

**Comment by jpl-jengelke** [comment]
The next PR will show the changes Closing this one for now.

**Pull request closed without merging**